### DISCUSSION TITLE
Help with SMTP Providers settings

### DISCUSSION THREAD

**Discussion by doncicuto** [zitadel/zitadel#6667] [Contributors]
As I'm working in #5503 I'd like to ask Zitadel team and fellow users if you can help me with default settings (known working configurations) for your favorite providers.

I can (and will) look for those settings if needed :smile:  but if you could help me, I'd need for your favorite providers:

- The smtp server host name (for example smtp.sendgrid.com)
- The default port for unencrypted connections (for example 587 for Sendgrid)
- The default port for TLS connections (for example 465 for Sendgrid)
- The default user/username if any (for example Sendgrid expects apikey)
- What is used as the SMTP Password (for example Sendgrid uses the API Key)

As I've no access to every provider so if you can share this information any help will be appreciated 

So far I'll try to add settings for Mailgun, Sendgrid, Postmark, Amazon SES, Google, Mailjet.... but if you add the settings here I can add more providers :smile: 

Thanks!

**Comment by hifabienne** [comment]
@fforootd @eliobischof I think you both tried out multiple providers for our cloud, maybe you still have the information

**Comment by doncicuto** [comment]
Yesterday I manage to talk with Postmark support and they were very helpful giving me a simple dev account for my testing purposes.

**Comment by doncicuto** [comment]
Mailgun
========
smtp server: smtp.mailgun.org
default port for unencrypted connections: 587
default port for TLS connections: 465
The default user/username: postmaster@YOURDOMAIN
SMTP Password: must be set by user

**Comment by relu91** [comment]
Hi @doncicuto thank you for working on https://github.com/zitadel/zitadel/issues/5503. You mentioned that you tried a bunch of SMTP servers, we are experiencing some problems with Google. It seems it always rejects user and pass (even if they are correct). Searching around it seems that 2FA is blocking direct login with user credentials. Do you have more insights on how you make it works on Zitadel ?

**Comment by doncicuto** [comment]
Good morning @relu91. If you want to use Gmail with a user/password have you created an App Password? Previously Google allowed to create settings for less secure passwords but more recently you need to create tha App Password and use it as the password for the mailbox: https://www.androidpolice.com/how-to-allow-less-secure-apps-in-gmail/

Use TLS
Host and port: smtp.gmail.com:587
User is your gmail address
Password would be the App Password

Would this help you?

**Comment by relu91** [comment]
Thank you! We tried disabling 2FA but without success. Fun fact, even if you disable 2FA it is still active and you don't have the option to create an App Password. You have first re-enable 2FA and **then** create an APP password. 

Anyhow issue solved 🥳 .

**Comment by doncicuto** [comment]
Cool @relu91 thanks for the feedback!. I'd like to add a template for Google Mail so it's good to know the workaround for the 2FA. I'm closing the discussion as it was an old discussion I opened for the #5503 issue but if you need any help you know where I am :smile

**Comment by dvl-numeez** [comment]
Hi everyone,

I'm currently running Zitadel locally using Docker and attempting to configure an SMTP provider using SendGrid.

I've double-checked the credentials and followed SendGrid's official SMTP guidelines, but I can't get past this following error.

**RpcError(INTERNAL): could not add smtp auth, check if both your user and password are correct, if they're correct maybe your provider requires an auth method not supported by ZITADEL (EMAIL-s9kfs)**

I have tried using TLS and without TLS 
Changing of PORT also did not work

Is there anything I'm missing? Does Zitadel currently support SendGrid's SMTP authentication method? Any help or guidance would be greatly appreciated.


Thanks in advance!
